app/src/main/java/com/mikhailovskii/weatherandroid/ui/choose_location/ChooseLocationActivity.kt

```diff
@@ -0,0 +1,48 @@
+package com.mikhailovskii.weatherandroid.ui.choose_location
+
+import android.content.Intent
+import android.os.Bundle
+import androidx.appcompat.app.AppCompatActivity
+import com.mikhailovskii.weatherandroid.R
+import com.mikhailovskii.weatherandroid.ui.main.MainActivity
+import kotlinx.android.synthetic.main.activity_choose_location.*
+
+class ChooseLocationActivity : AppCompatActivity(), ChooseLocationContract.ChooseLocationView {
+
+    private val presenter = ChooseLocationPresenter()
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_choose_location)
+
+        presenter.attachView(this)
+
+        next_btn.setOnClickListener {
+            when {
+                country_tiet.text?.isBlank() == true -> {
+                    country_til.error = getString(R.string.fill_the_field)
+                    return@setOnClickListener
+                }
+                city_tiet.text?.isBlank() == true -> {
+                    city_til.error = getString(R.string.fill_the_field)
+                    return@setOnClickListener
+                }
+                else -> {
```

Comment:
@SergeiMikhailovskii same as in your DiffUtils case. Your `else` statement now handles the cases when `city_tiet.text == null` or when `country_tiet.text == null` which, I guess, is not what you expected to have writing your code.

Reply:
try to use kotlin decompiler in android studio and see how the same code would look in java to understand how question mark (nullability check) works.

Reply:
@anton-novikau is there any way to get `null` here? I have the following logic: even if the user didn't enter anything to fields, we get `tiet.text?.isBlank() == true` here, but not null. Or am I wrong?

Reply:
@SergeiMikhailovskii since `TextView.getText()` is declared as nullable, there is always a possibility to have a `null` as a return value. Probably on some devices it works, but it's android, so everything may happen. Leave it on your own risk